authors/apps/articles/serializers.py

```diff
@@ -30,32 +42,37 @@ def to_representation(self, instance):
         article_rep['read_time'] = str(readtime.of_html(article_rep['body']))
         # return the article's details along with it's read time
         return article_rep
+    
 
     def create(self, validated_data):
         """
         Create and return a new `Article` instance, given the validated data.
         """
-        return Articles.objects.create(**validated_data)
+        tags = [tag.lower() for tag in validated_data.pop('tags')]
+        instance = super().create(validated_data)
+        instance.tags.set(*tags)
+        return instance
 
     def update(self, instance, validated_data):
         """
         Update and return an existing `Article`, given the validated data.
         """
+        tags = validated_data.pop('tags', instance.tags.all())
         instance.title = validated_data.get('title', instance.title)
         instance.body = validated_data.get('body', instance.body)
-        instance.description = validated_data.get(
-            'description',
-            instance.description)
+        instance.description = validated_data.get('description', instance.description)
+        instance.tags.set(*tags)
+
```

Comment:
This replaces the tags list rather than update it, ie. if I add/update a single tag to an article the tags original list will be replaced by the new single element list. Is this intended?

Reply:
if i may, with regards to whether one is able to update an article on his branch, it is possible,

Reply:
It wasn't intended.
I have implemented this feedback and confirmed that I can update the tags without removing the previous ones.